src/lib/helpers.ts

```diff
@@ -385,76 +379,25 @@ export function alphaNumericSort(
   }
 }
 
-/**
- * Function  to get the object down in object hierarchy that have given property as a member property
- * @param findProperty - Property to search
- * @param obj -  Object to look for the property
- */
-export function getParentObjectForField(
-  findProperty: string,
-  obj: StringKeyedProps
-): Object | undefined {
-  if (Object.keys(obj).includes(findProperty)) {
-    return obj;
-  }
-  let ret = undefined;
-  for (let prop in obj) {
-    if (obj.hasOwnProperty(prop) && typeof obj[prop] === "object") {
-      ret = getParentObjectForField(findProperty, obj[prop]);
-      if (ret) {
-        return ret;
-      }
-    }
-  }
-  return ret;
-}
-
-/**
- * Get the value of the given property from an object or from nested objects
- * @param object - Object to look for the property
- * @param propertyName - Property to search for
- * @param identiferForProp - This is useful when different object types hold property of same name
- * @example - Work object contains 'workType' and 'project' objects with member property 'name'.
- * In such cases,to access the name property of workType, the propertyName can be given as "workType" and  identiferForProp as "name"
- **/
+/**Get value for a property field from an object using the given path
+ * @param obj Object to search for
+ * @param propertyPath Property names to traverse the object to reach the required search property field
+ * @example  To get 'name' field value in a 'work' object - 'work'  holds a property 'workType' which is an object with 'name' field
+ *            the path should be ["workType","name"]
+ ***/
 export const getPropertyValue = (
-  object: Object,
-  propertyName: string,
-  identiferForProp?: string
-): string | number => {
-  //Get the object containing the propertyName
-  const parentObjForProperty:
-    | StringKeyedProps
-    | undefined = getParentObjectForField(propertyName, object);
-  if (!parentObjForProperty) return "";
-
-  //Get value for property name given
-  const propValue = parentObjForProperty[propertyName];
-  if (typeof propValue !== "object") {
-    return propValue;
+  obj: StringKeyedProps,
+  propertyPath: string[]
+): number | string => {
+  let propValue: any = obj;
+  for (let indx = 0; indx < propertyPath.length; indx++) {
+    const val = propValue[propertyPath[indx]];
+    if (!val) break;
```

Comment:
If this break occurs, what are we expecting getPropertyValue to end up returning?

Reply:
In that case the return value will be an empty string. Do you think it needs to be more explicit?